add blog post 'Falcoctl: install and manage your rules and plugins' (#827)
falcosecurity/falco-website: content/en/blog/falcoctl-install-manage-rules-plugins.md/index.md

---
title: "Falcoctl: The basic usages"
linktitle: "Falcoctl: The basic usages"
date: 2023-02-16
author: Thomas Labarussias
slug: falcoctl-basic-usages
---

> Since the launch of the plugin framework in January 2022, our adopters have requested an out-of-the-box solution to manage the lifecycle of rules (installation, updates), and a how to guide to smoothly install the [plugins](https://falco.org/docs/plugins/).

Reviewer: ```suggestion
Since the launch of the plugin framework in January 2022, our adopters have requested an out-of-the-box solution to manage the lifecycle of rules (installation, updates). We heard your request and also created a guide to help you smoothly install the [plugins](https://falco.org/docs/plugins/).
```

- [Glossary](#glossary)
- [Use falcoctl locally](#use-falcoctl-locally)
  - [Installation](#installation)
  - [Index](#index)
  - [Configuration](#configuration)
  - [Search for artifacts](#search-for-artifacts)
  - [Installation of artifacts](#installation-of-artifacts)
    - [Installation of Rules](#installation-of-rules)
    - [Installation of Plugins](#installation-of-plugins)
    - [Follow artifacts](#follow-artifacts)
      - [Systemd service](#systemd-service)
- [Use falcoctl in Kubernetes with Helm](#use-falcoctl-in-kubernetes-with-helm)
  - [Add the official Helm repo](#add-the-official-helm-repo)
  - [Configure the installation and followed by the artifacts](#configure-the-installation-and-followed-by-the-artifacts)
- [Disable falcoctl in the chart](#disable-falcoctl-in-the-chart)
- [Conclusion](#conclusion)

## Glossary

Before diving into all the details, here's a glossary of the key words to understand how `falcoctl` works. An exhaustive list can be found in the [README](https://github.com/falcosecurity/falcoctl/blob/main/README.md).

- `Artifact`: An artifact is an element `falcoctl` can manipulate, right now, only `rules` and `plugins` are considered.

> - `Index`: A .yaml file containing a list of available artifacts with their registries and repositories. An automatic generic index is provided for official artifacts, see [here](https://github.com/falcosecurity/falcoctl/tree/gh-pages).

Reviewer 2: ```suggestion
- `Index`: A .yaml file containing a list of available artifacts with their registries and repositories. The tool's default configuration contains an index file pointing to the officially supported artifacts from the `falcosecurity` organization, see [here](https://github.com/falcosecurity/falcoctl/tree/gh-pages). Users can also maintain their own index files, pointing to registries and repositories containing custom rulesfiles and plugins.
```

- `Registry`: A registry stores the artifacts, that `falcoctl` understands in regards to the [OCI standard](https://opencontainers.org/), any compliant OCI can be used. The official registry uses [Github Packages](https://github.com/orgs/falcosecurity/packages).
- `Repository`: similar to containers, it belongs to a registry and contains one or more versions (tags) of an artifact.

## Use falcoctl locally

The following steps explain the basic concepts used to install `falcoctl` locally.

### Installation

For a Linux amd64 system:
```shell
LATEST=$(curl -sI https://github.com/falcosecurity/falcoctl/releases/latest | awk '/location: /{gsub("\r","",$2);split($2,v,"/");print substr(v[8],2)}')
curl --fail -LS "https://github.com/falcosecurity/falcoctl/releases/download/v${LATEST}/falcoctl_${LATEST}_linux_amd64.tar.gz" | tar -xz
sudo install -o root -g root -m 0755 falcoctl /usr/local/bin/falcoctl
```

> Other environments are described in the [README](https://github.com/falcosecurity/falcoctl#install-falcoctl-manually)

> You can now check if the installation went well.

Reviewer: ```suggestion
Now you are able to confirm that the installation went well.
```

Check the version:
```shell
❯ falcoctl version

Client Version: 0.4.0
```
Print the help:
```shell
❯ falcoctl --help

     __       _                _   _
    / _| __ _| | ___ ___   ___| |_| |
   | |_ / _  | |/ __/ _ \ / __| __| |
   |  _| (_| | | (_| (_) | (__| |_| |
   |_|  \__,_|_|\___\___/ \___|\__|_|
									
	

> The official CLI tool for working with Falco and its ecosystem components

Reviewer: ```suggestion
The official CLI tool for working with Falco and its ecosystem components.
```

Usage:
  falcoctl [command]

Available Commands:
  artifact    Interact with Falco artifacts
  completion  Generate the autocompletion script for the specified shell
  help        Help with any command
  index       Interact with index
  registry    Interact with OCI registries
  tls         Generate and install TLS material for Falco
  version     Print the falcoctl version information

Flags:
      --config string     config file to be used for falcoctl (default "/etc/falcoctl/falcoctl.yaml")
      --disable-styling   Disable output styling such as spinners, progress bars and colors. Styling is automatically disabled if not attached to a tty (default false)
  -h, --help              help for falcoctl
  -v, --verbose           Enable verbose logs (default false)

Use "falcoctl [command] --help" for more information about a command.

```

### Index

> Before being able to download and install artifacts, we need to configure an index, we provide one for all [official artifacts](https://falcosecurity.github.io/falcoctl/index.yaml) (plugins and rules):

Reviewer: ```suggestion
Before being able to download and install artifacts, we need to configure an index, we provide one for all [official artifacts](https://falcosecurity.github.io/falcoctl/index.yaml), plugins and rules:
```

> Before being able to download and install artifacts, we need to configure an index, we provide one for all [official artifacts](https://falcosecurity.github.io/falcoctl/index.yaml), plugins and rules:  

Reviewer 2: Not completely true in my opinion: if you want to install and follow artifacts, the tool has its own default config containing our official index. On the other hand, this section could be used to show how a custom index can be configured, but normal users will never use this.

Author: The configuration and setup of a custom index is for a next blog post.

Reviewer 2: Yeah so maybe I was thinking we can skip this `Index` section?

Author: I see that as an introduction to make people understand the concepts.

```yaml
...
falcoctl:
  artifact:
    install:
      enabled: true
    follow:
      enabled: true
  config:
    indexes:
    - name: falcosecurity
      url: https://falcosecurity.github.io/falcoctl/index.yaml
    artifact:
      allowedTypes:
        - rulesfile
        - plugin
      install:
        resolveDeps: true
        refs: [k8saudit-rules:0.5, falco-rules:0]
      follow:
        refs: [k8saudit-rules:0.5, falco-rules:0]
        every: 6h
...
```

> The `resolveDeps: true` avoids listing the plugin at install, and will be automatically installed with the rules.

Here's the final command to bootstrap a very basic installation with these settings:
```shell
helm install falco -n falco -f values.yaml falcosecurity/falco --create-namespace
```

## Disable falcoctl in the chart

If for some reason you don't want to use `falcoctl` to manage artifacts, you can disable its installation by following these steps:

```shell
helm install falco \
    --set falcoctl.artifact.install.enabled=false \
    --set falcoctl.artifact.follow.enabled=false
```

## Conclusion

> In this blog post we saw the basics to use `falcoctl` locally and in our Kubernetes clusters with `helm`. It helps for sure to manage the lifecycle of the rules and the plugins, moreover its development is strongly active and more and more features and enhancements will be added.

Reviewer: ```suggestion
In this blog post we learned some of the basic `falcoctl` usages that we can run locally and in our Kubernetes clusters with `helm`. It helps to manage the lifecycle of the rules and the plugins with an out of the box solution like falcoctl`. Stay tuned as the plugin ecosystem is growing and we are seeing the development of more features and enhancements.
```

> More posts will also be out soon to detail more advanced usages, like using private registries or creating your own artifacts. Stay tuned!

Reviewer: ```suggestion
More posts will also be out soon to detail more advanced usages, such as using private registries and creating your own artifacts. See you soon!
```
